source/jormungandr/jormungandr/utils.py

```diff
@@ -534,6 +534,48 @@ def get_pt_object_coord(pt_object):
     return coord
 
 
+def is_olympic_poi(pt_object, instance):
+    if pt_object.embedded_type != type_pb2.POI:
+        return False
+    if not (hasattr(pt_object.poi, 'properties') and pt_object.poi.properties):
+        return False
+    olympic_site = next(
+        (
+            p
+            for p in pt_object.poi.properties
+            if p.type == instance.olympics_forbidden_uris.poi_property_key
+            and p.value == instance.olympics_forbidden_uris.poi_property_value
+        ),
+        None,
+    )
+    if olympic_site:
+        return True
+    return False
+
+
+def is_olympic_site(entry_point, instance):
+    if not entry_point:
+        return False
+    if is_olympic_poi(entry_point, instance):
+        return True
+    if entry_point.embedded_type == type_pb2.ADDRESS:
+        if not (hasattr(entry_point.address, 'within_zones') and entry_point.address.within_zones):
+            return False
+        for within_zone in entry_point.address.within_zones:
+            if is_olympic_poi(within_zone, instance):
+                return True
+    return False
+
+
+def get_first_or_last_pt_section(journey, first=True):
```

Comment:
This is weird naming. Looks like you will get `first` or `last` randomly. If you get it from a parameter, why not do 2 functions `get_first_pt_section` and `get_last_pt_section`?

Another idea, would be to rename the function to something like `get_olympic_site_pt_section` which describes better what you're trying to do, isn't it? You could even take 2 parameters to decide which one should be returned, the origin (`None` or something) and the destination (`None` or something).

Reply:
:+1:

Reply:
Done